pkg/catalogsourceconfig/factory.go

```diff
@@ -39,10 +39,14 @@ type phaseReconcilerFactory struct {
 
 func (f *phaseReconcilerFactory) GetPhaseReconciler(log *logrus.Entry, csc *v1alpha1.CatalogSourceConfig) (Reconciler, error) {
```

Comment:
Shouldn't  `if !csc.ObjectMeta.DeletionTimestamp.IsZero() {` be at the top?

Reply:
Probably true, but my thinking is that such a change is outside the scope of this pr.

Reply:
Oh, I actually see what you're saying. We should also probably check that we shouldn't go to the deleted phase first.